models/readme.md

```diff
@@ -1 +1,164 @@
+- `popular_on_huggingface`: hundreds of the most-downloaded models from the [Huggingface models repository](https://huggingface.co/models).
+- `selftest`: a small corpus with small models that can be used for testing out the MLAgility tools.
+- `torch_hub`: a variety of models, including many image classification models, from the [Torch Hub repository](https://github.com/pytorch/hub).
+- `torchvision`: image recognition models from the [`torchvision` library](https://pytorch.org/vision/stable/index.html).
+  - _Note_: the `torchvision` library also includes many image classification models, but we excluded them to avoid overlap between our `torch_hub` and `torchvision` corpora.
+- `transformers_keras`: Transformer models from the [Huggingface `transformers` library](https://huggingface.co/docs/transformers/index), expressed in the Keras framework.
+- `transformers_pytorch`: Transformer models from the [Huggingface `transformers` library](https://huggingface.co/docs/transformers/index), expressed in the PyTorch framework.
+
+## Running the Benchmark
+
+### Prerequisites
+
+Before running the benchmark we suggest you familiarize yourself with the [`benchit` CLI tool](https://github.com/groq/mlagility/blob/main/docs/benchit_user_guide.md) and some of the [`benchit` CLI examples](https://github.com/groq/mlagility/tree/main/examples/cli).
+
+You must also run the following command to install all of the models' dependencies into your Python environment.
+
+`pip install -r models/requirements.txt`
+
+### Benchmarking Commands
+
+Once you have fulfilled the prerequisites, you can evaluate one model from the benchmark with a command like this:
+
+```
+cd MLAGILITY_ROOT/models # MLAGILITY_ROOT is where you cloned mlagility
+benchit benchmark linear.py --search-dir selftest
+```
+
+You can also evaluate an entire corpus with a command like this:
+```
+cd MLAGILITY_ROOT/models # MLAGILITY_ROOT is where you cloned mlagility
+benchit benchmark --all --search-dir selftest
+```
+
+You can aggregate all of the benchmarking results from your `mlagility cache` into a CSV file with:
+
+```
+benchit report
+```
+
+If you want to only report on a subset of models, we recommend benchmarking those models into a specific cache directory:
+
+```
+# Benchmark the models into a specific cache directory
+benchit benchmark --all --search-dir selftest --cache-dir selftest_results
+
+# Report the results from the `selftest_results` cache
+benchit report --cache-dir selftest_results
+```
+
+## Model Template
+
+Each model in the MLAgility benchmark follows a template to keep things consistent. This template is meant to be as simple as possible. The models themselves should also be instantiated and called in a completely vendor-neutral manner.
+
+### The model.py File
+
+Each model in the MLAgility benchmark is hosted in a `model.py` file. This file is a Python script that must meet the following requirements:
```

Comment:
Is the idea to only allow scripts in mlagility if that script only instantiates a single model?

Reply:
Nope, reworked the definition